core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,1002 @@
+                    choices = (
+                    value.interaction.customization_args['choices'].value)
+                    min_value = (
+                        value.interaction.customization_args
+                        ['minAllowableSelectionCount'].value)
+                    max_value = (
+                        value.interaction.customization_args
+                        ['maxAllowableSelectionCount'].value)
+                    for rule_spec in answer_group.rule_specs:
+                        rule_spec_index = str(
+                            answer_group.rule_specs.index(rule_spec))
+                        if rule_spec.rule_type == 'Equals':
+                            if (len(rule_spec.inputs['x']) < min_value or
+                                len(rule_spec.inputs['x']) > max_value):
+                                item_selec_interaction_values.append(
+                                    'Selected choices of rule {rule} '.format(
+                                        rule=rule_spec_index) +
+                                    'of answer group {ans_group} '.format(
+                                        ans_group=answer_group_index) +
+                                    'either less than min_selection_value ' +
+                                    'or greter than max_selection_value.')
+
+            if value.interaction.id == 'MultipleChoiceInput':
+                choices = (
+                    value.interaction.customization_args['choices'].value)
+                # Validates if the choices are less than 4.
+                if len(choices) < 4:
+                    mc_interaction_invalid_values.append(
+                        'There should be atleast 4 choices'
+                        + ' found ' + str(len(choices)))
+                # Validates if the choice is empty or duplicate.
+                seen_choices = []
+                choice = None
+                choice_empty = False
+                choice_duplicate = False
+                for choice in choices:
+                    if choice.html == '<p></p>':
+                        choice_empty = True
+                    if choice.html in seen_choices:
+                        choice_duplicate = True
+                    seen_choices.append(choice.html)
+                if choice_empty and choice is not None:
+                    mc_interaction_invalid_values.append(
+                        'There should not be any empty' +
+                        ' choices')
+                if choice_duplicate and choice is not None:
+                    mc_interaction_invalid_values.append(
+                        'There should not be any duplicate' +
+                        ' choices')
+                # Validates if all have feedback and still have deflt outcome.
+                if (len(choices) == len(value.interaction.answer_groups)
+                    and value.interaction.default_outcome is not None):
```

Comment:
```suggestion
                if (
                        len(choices) == len(value.interaction.answer_groups) and
                        value.interaction.default_outcome is not None
                ):
```

Why is this condition correct? Can `answer_groups` have no feedback?

Reply:
This check is required in case when all the choices have the feedback defined in the answer group and still have an option for default outcome feedback, we would not require any default feedback as all choices have separate feedback.
Thanks

Reply:
But I don't understand how the `len` of `answer_groups` checks that the feedback is provided, or does `answer_groups` just contain the feedback?

Reply:
So each answer group has rule specs and each rule spec has feedback attached to it. And it is presumed that if we have the number of answer groups equal to the choices then it would explain that we have provided feedback for each choice we have.
This method is also followed in the frontend validation.
Thanks

Reply:
Ah I see. So we assume that there will be at least some feedback in each answer group.

Reply:
Yes